language/move-lang/src/hlir/translate.rs

```diff
@@ -1168,16 +1194,30 @@ fn exp_impl(
                     .map(|(e, (f, bt))| (f, bt, e))
                     .collect()
             } else {
-                let mut fields = (0..decl_fields.len()).map(|_| None).collect::<Vec<_>>();
+                let num_fields = decl_fields.as_ref().map(|m| m.len()).unwrap_or(0);
+                let mut fields = (0..num_fields).map(|_| None).collect::<Vec<_>>();
                 for (decl_idx, f, _exp_idx, bt, tf) in texp_fields {
+                    // Might have too many arguments, there will be an error from typing
```

Comment:
This comment is making me wonder if eventually we should have a way of specifying  "nonblocking error X will result in blocking errors A, B, or C" later on in compilation.

Reply:
Yeah... I've thought about something like this but I think they would be very hard to maintain. Maybe would be good to check that there is a given error in the diags though